crypto/rand/drbg_lib.c

```diff
@@ -829,7 +829,7 @@ static int drbg_bytes(unsigned char *out, int count)
     if (drbg == NULL)
         return 0;
 
-    CRYPTO_THREAD_write_lock(drbg->lock);
+    RAND_DRBG_lock(drbg);
```

Comment:
Formally, one would have to check the return value when taking the lock. But this does not seem to be common practice in OpenSSL source code. Thoughts?

```
msp@msppc:~/src/openssl$ grep -rnH  CRYPTO_THREAD_write_lock * | grep -e if -e =
crypto/threads_pthread.c:160:    if (!CRYPTO_THREAD_write_lock(lock))
crypto/threads_pthread.c:199:    if (!CRYPTO_THREAD_write_lock(lock))
```